server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -237,10 +239,12 @@ export function configureWebSocketServices(
             const version = selectProtocolVersion(connectVersions);
             if (!version) {
                 // eslint-disable-next-line prefer-promise-reject-errors
-                return Promise.reject(
-                    `Unsupported client protocol.` +
+                return Promise.reject({
+                    code: 400,
+                    message: `Unsupported client protocol. ` +
                     `Server: ${protocolVersions}. ` +
-                    `Client: ${JSON.stringify(connectVersions)}`);
+                    `Client: ${JSON.stringify(connectVersions)}`,
+                });
             }
```

Comment:
I believe this should be non retryable as well. If you don't send the correct protocol version, you should definitely update your app, which means you need to load the app again?

Reply:
Agreed. 400 is non-retryable from the driver perspective ([r11s-driver/errorUtils.ts](https://github.com/microsoft/FluidFramework/blob/main/packages/drivers/routerlicious-driver/src/errorUtils.ts#L67-L68)) because it's a "Bad Request, The server could not understand the request due to invalid syntax." According to [MDN's 400 Bad Request docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400), "the client should not repeat this request without modification." 

ODSP follows similar logic for 400s ([utils/odsp-doclib-utils](https://github.com/microsoft/FluidFramework/blob/main/packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts#L129-L130)).

Is there an alternative error code you'd suggest?

Reply:
Great! As long as its not retryable, we are good.